src/identity/manager.rs

```diff
@@ -32,16 +37,141 @@ impl fmt::Display for Identity {
 #[derive(Clone)]
 pub struct SecretManager {
     client: CaClient,
+    cache: Arc<RwLock<HashMap<Identity, watch::Receiver<Option<tls::Certs>>>>>,
 }
 
 impl SecretManager {
     pub fn new(cfg: crate::config::Config) -> SecretManager {
-        let client = CaClient::new(cfg.auth);
-        SecretManager { client }
+        let caclient = CaClient::new(cfg.auth);
+        let cache: HashMap<Identity, watch::Receiver<Option<tls::Certs>>> = Default::default();
+        SecretManager {
+            client: caclient,
+            cache: Arc::new(RwLock::new(cache)),
+        }
+    }
+
+    pub async fn refresh_handler(
+        id: Identity,
+        mut ctx: SecretManager,
```

Comment:
Need to clone self so that the compiler is not scared of self's lifetime being less than task.  Steven's suggestion of passing the necessary fields individually is probably what I'll do, but it's a bit messy.